controllers/volumereplicationgroup_controller.go

```diff
@@ -607,17 +607,17 @@ func (v *VRGInstance) processForDeletion() (ctrl.Result, error) {
 	return ctrl.Result{}, nil
 }
 
-// reconcileVRsForDeletion cleans up VR resources managed by VRG and also cleans up changes made to PVCs
+// processPVCsForDeletion cleans up VR resources managed by VRG and also cleans up changes made to PVCs
 // TODO: Currently removes VR requests unconditionally, needs to ensure it is managed by VRG
-func (v *VRGInstance) reconcileVRsForDeletion() bool {
+func (v *VRGInstance) processPVCsForDeletion() bool {
 	requeue := false
 
 	for idx := range v.pvcList.Items {
 		pvc := &v.pvcList.Items[idx]
 		pvcNamespacedName := types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}
 		log := v.log.WithValues("pvc", pvcNamespacedName.String())
 
-		requeueResult, skip := v.preparePVCForVRProtection(pvc, log)
+		requeueResult, skip := v.prepareToProtectEligiblePVC(pvc, log)
```

Comment:
> I think this function renaming should change, to drop eligible from the name. It's return values suggest appropriate eligibility actions for now.

Just for context, this method was originally named `preparePVCForVRProtection()` that in turn invoked the `protectPVC()` method.

Let me try to explain how I ended up adding the `Eligible` qualifier. Protection of PVC involves two main tasks: enabling replication via VolRep (VR) and uploading the PV resource to S3 (and in future, also uploading the snapshot resources, etc).  Given the main change I was making in the PR (moving the S3 upload and moving the protected annotation to a later point in the flow), I thought that the name `protectPVC()` no longer is the right fit for this method and can be even misleading, as it will be merely adding a finalizer and retaining the PV.  In my mind, the addition of the finalizer is a preparation not only required to clean up the VR but also to clean up S3. Hence, I thought this method is better named `prepareToProtectPVC()`.  I followed the existing imperative style of naming the method `protectPVC()` and renamed it to `prepareToProtectPVC()`. Given this, I had to differentiate the caller `preparePVCForVRProtection()` to `prepareToProtectEligiblePVC()`, which was doing eligibility checks. 

> The function specifically prepares a PVC to be replicated by a VR, as some of the conditions within the function are VR specific and not generic.

Please correct me if I'm mistaken, but with the change I'm making, I think the preparation phase is no longer limited to VR (which is why I got rid of the VR qualifier). To me, the preparation step of adding a finalizer is required to clean up S3 data.  I believe the preparation step is being used as a two phase commit operation: finalizer at the start and annotation at the end.  If the finalizer is present, it indicates that we have started the series of protection steps, like s3 upload, VR creation, etc., but without the presence of the annotation, we don't know which of those steps have completed successfully. During deletion of VRG, if the finalizer is present on PVC we need to clean up S3,  delete the VR, etc. Hence, this step is no longer VR specific in my mind.
 
In its present state, preparing the PVC involves checking if the PVC is eligible [done by `prepareToProtectEligiblePVC()`] which in turn does the preparation (initial steps of protecting the PVC) by adding a finalizer and retaining the PV [done by calling `prepareToProtectPVC()`]. I recall that there was some pending decision on whether PV retention is required or not. If we decide in future to not retain the PV, then `prepareToProtectPVC()` will be just adding the finalizer.

> Should it be renamed here, or should this call a generic preparePVCForReplication and that calls a protectPVCForVR is something we can do later? I would suggest we rename this as preparePVCForReplication

In my mind, the term "replication" is about protecting the PV backend data via VR and doesn't include backing up the PV cluster data (or resource).  I don't know if that's also how you use the term replication here. Whereas, as I outlined above, I believe this method is preparing the PVC for both: replication of PV backend data and backup of PV cluster data; hence, protection seems more appropriate, whereas replication seems more narrow.  

Thanks for the openness you all show by giving the renaming commit a fair look.  I understand that there's no good time to change names, because all those who are familiar with the code have to unlearn what they know.  Given that the change I was making (moving the S3 upload and moving the protected annotation to a later point in the flow) was disruptive enough, there is bound to be some relearning that is required.  Hence, I used it as an opportunity to look at what a method was being done to come up with descriptive names that match what it is doing today (not what it was doing in the past).  Hopefully, the new names reinforce the changed world better.  But, I'm happy to consider different names or keep the old names if that seems to be a better fit for what the method is doing today.

It would be easier if we all could read what's on each other's mind, but until then, we have to explain with long notes! :-)

Reply:
I think abstracting this out to non-VR at present is too early, I would hence suggest we do something like `preparePVCForProtection` so that subsequent methods are protectPVCAsPrimary|Secondary, IOW prepare and then protect.

With the prepare -> protect -> preserve (for S3 backups) forming the 2 action steps?

Reply:
I can see the following method rename here:
`preparePVCForVRProtection` -> `preparePVCForProtection`
`protectPVC` -> `prepareEligiblePVCForProtection`

Reply:
I can't help but wonder about how the new names have reversed and yet makes sense. The power of diverse thought! I like it; will use these two.

Regarding protect vs preserve... VRG is in the job of protecting PVCs, which consists of two equally important parts:
1. Protecting the PV Contents: PV backend volume for now, PV snapshot contents in future (this is `what` it does). VRG creates VR CRs to enable replication of the storage volume (this is `how` it protects PV Contents).
2. Protecting the PV Cluster Data, aka PV related k8s objects or resources: PV for now, and snapshots in future (this is `what` it does). VRG backs up the PV Cluster Data to S3 (this is `how` it protects PV Cluster state).

If either of the above two fails, then the PV is not protected and the VRG state is not suitable for relocation.

Reply:
> Regarding protect vs preserve... VRG is in the job of protecting PVCs, which consists of two equally important parts:
> 
>     1. Protecting the PV Contents: PV backend volume for now, PV snapshot contents in future (this is `what` it does). VRG creates VR CRs to enable replication of the storage volume (this is `how` it protects PV Contents).
> 
>     2. Protecting the PV Cluster Data, aka PV related k8s objects or resources: PV for now, and snapshots in future (this is `what` it does). VRG backs up the PV Cluster Data to S3 (this is `how` it protects PV Cluster state).
> 
> 
> If either of the above two fails, then the PV is not protected and the VRG state is not suitable for relocation.

I agree on the "what" and "how", all I am saying at present is that both of the "how" can change (protect PV cluster data may change to something else (replication cookies?) and instead of using VR we use other replication methods) and it is too early to confuse us with an abstraction that makes better sense in the future.

IOW I would like to reason with code as doing VR stuff and then doing PV stuff, than doing backend data stuff (which means VR) and doing cluster data (which means PV backups).

Hence the resistance to rename these as broadly as you suggest. We may/will end up here in a hopefully not too distant future, but let's wait till then.

Reply:
I understand.

Reply:
> We may/will end up here in a hopefully not too distant future, but let's wait till then.

I withdrew the renaming commit and kept the commits only to fixes.